client-app/shared/fulfillmentCenters/components/branches-dialog.vue

```diff
@@ -0,0 +1,206 @@
+          </button>
+
+          <button
+            class="self-center flex items-center justify-center ml-2.5 w-7 h-7 rounded shadow-t-mds bg-white"
+            @click="clearSelection"
+          >
+            <svg class="text-primary" width="16" height="16">
+              <use href="/static/images/clear.svg#main"></use>
+            </svg>
+          </button>
+        </template>
+      </div>
+      <div class="px-6 py-3 border-b">
+        <BranchSearch :model-value="searchInput" @update:input="search($event)" />
+      </div>
+      <div>
+        <template v-for="(branch, index) in branches">
+          <BranchItem
+            v-if="(showSelectedBranchesMobile && isBranchSelected(index)) || !showSelectedBranchesMobile"
+            :branch="branch"
+            :is-text-truncate-enabled="false"
+          >
+            <VcCheckbox class="mr-3 cursor-pointer" v-model="selectedBranchesIds" :value="branch.id"> </VcCheckbox>
+          </BranchItem>
+        </template>
+      </div>
+    </div>
+    <!-- MOBILE content END -->
+
+    <template #actions="{ close }">
+      <div
+        class="relative flex-grow flex items-center justify-between -mt-4 pt-4 -mx-6 px-10 gap-5 shadow-t-lgs bg-white sm:shadow-none sm:mt-0 sm:px-6 sm:pt-0 sm:gap-auto"
+      >
+        <VcButton
+          kind="secondary"
+          class="uppercase basis-0 flex-grow sm:basis-auto sm:flex-grow-0 sm:px-4"
+          is-outline
+          @click="close"
+        >
+          {{ $t("shared.catalog.branches_dialog.cancel_button") }}
+        </VcButton>
+
+        <VcButton
+          class="uppercase basis-0 flex-grow sm:basis-auto sm:flex-grow-0 sm:px-5 sm:min-w-[9rem]"
+          :is-disabled="isSaveButtonDisabled"
+          @click="
+            save();
+            close();
+          "
+        >
+          {{ $t("shared.catalog.branches_dialog.ok_button") }}
+        </VcButton>
+      </div>
+    </template>
+  </VcPopup>
+</template>
+
+<script setup lang="ts">
+import { computed, ref, watch } from "vue";
+import { breakpointsTailwind, useBreakpoints, useLocalStorage } from "@vueuse/core";
+import BranchItem from "./branch-item.vue";
+import BranchSearch from "./branch-search.vue";
+import { useFulfillmentCenters, IFulfillmentCenter, FFC_TEMP_LOCAL_STORAGE_NAME } from "@/shared/fulfillmentCenters";
+
+const { loading, loadFulfillmentCenters, fulfillmentCenters } = useFulfillmentCenters();
+const breakpoints = useBreakpoints(breakpointsTailwind);
+const isMobile = breakpoints.smaller("sm");
+const isMobileSidebar = breakpoints.smaller("lg");
+
+const showSelectedBranches = ref(false);
+const showSelectedBranchesMobile = ref(false);
+const searchInput = ref<string>("");
+const branches = computed(() => fulfillmentCenters.value.filter((item) => searchFilter(item)));
+const selectedBranchesIds = ref<string[]>([]);
+const selectedBranchesIdsTemp = useLocalStorage<string[]>(FFC_TEMP_LOCAL_STORAGE_NAME, []);
```

Comment:
don't use the temp localstorage value. Use event to pass result data. And operate result data in place.